sympy/series/limitseq.py

```diff
@@ -92,7 +92,7 @@ def dominant(expr, n):
 
     sympy.series.limitseq.dominant
     """
-    terms = Add.make_args(expr.expand(func=True))
+    terms = Add.make_args(expr.doit().expand(func=True))
```

Comment:
Why is this needed?

Reply:
Otherwise `dominant(n + Sum(k, (k, 0, n)), n)` wouldn't evaluate -- see OP. But I wonder if the input or None should just be returned if the expression didn't meet expectations. Not sure how be to introspect to know that, however. Perhaps if there are any objects with limits containing the series variable as a free symbol?

Reply:
I was stuck with an evaluation which was directly related to some issue or was inspired by some issue which involved a `Sum` expression inside `limit` . After some debugging I had realized that there's nothing wrong except this part which would be required to fix that . I'm forgetting exactly what was the question, I was trying to tackle ...... now that it's been some time since I made the Pr and add 2-3 weeks to that, when I tried that specific questions . I'll try to recall !

But anyways, I thought `doit`  would perform better in the general cases as in we have cases like this and `Sum` too returns None
```
>>> dominant(n**3 + n**2, n)
n**3
>>> dominant(n**3 + Product(n, (k, 1, 2)), n) 
RecursionError: maximum recursion depth exceeded while calling a Python object
```